packages/core/src/types.tsx

```diff
@@ -205,7 +205,13 @@ type NavigationHelpersCommon<
           name: RouteName;
           key?: string;
           params: ParamList[RouteName];
-          merge?: boolean;
+          merge?: false;
+        }
+      | {
+          name: RouteName;
+          key?: string;
+          params: Partial<ParamList[RouteName]>;
```

Comment:
The whole `params` could also be omitted when `merge` is `true`, like:

```suggestion
          params?: Partial<ParamList[RouteName]>;
```